<!DOCTYPE html>
<html lang="en">

  <head>
  <meta charset="UTF-8">
  <link href="http://gmpg.org/xfn/11" rel="profile">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Comparative RNA-seq &middot; Noah Reid
    
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="//public/css/poole.css">
  <link rel="stylesheet" href="//public/css/syntax.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="//public/apple-touch-icon-precomposed.png">
  <link rel="shortcut icon" href="//public/favicon.ico">

  <!-- RSS -->
  <link rel="alternate" type="application/atom+xml" title="Noah Reid" href="//atom.xml">
</head>


  <body>

    <div class="container content">
      <header class="masthead">
        <h3 class="masthead-title">
          <a href="//" title="Home">Noah Reid</a>
          <small></small>

	  
	  	&nbsp;&nbsp;&nbsp;
	  	<small><a href="/about/">About</a></small>
	  
	  	&nbsp;&nbsp;&nbsp;
	  	<small><a href="/research/">Research</a></small>
	  
	  	&nbsp;&nbsp;&nbsp;
	  	<small><a href="/publications/">Publications</a></small>
	  
	  	&nbsp;&nbsp;&nbsp;
	  	<small><a href="/software/">Software</a></small>
	  
	  	&nbsp;&nbsp;&nbsp;
	  	<small><a href="/archive">Archive</a></small>
	  

        </h3>
      </header>

      <main>
        <article class="post">
  <h2 class="post-title">Comparative RNA-seq</h2>
  <time datetime="2015-03-03T00:00:00-05:00" class="page-date">03 Mar 2015</time>
  <p>I just dove into a really cool project looking at patterns of gene expression underlying osmotic adaptation in a phylogenetic comparative context. I think this work is really, really cool, but from the very start, we’re faced with problems that I’m not sure how to deal with.</p>

<p>The choice I’m currently mulling is whether we should try to assemble transcriptomes de novo for each species, or map them all back to a common reference. If we do reference mapping, things are computationally simpler and a larger gene set will ultimately be evaluated, but it seems that there must be some biases associated with the mapping, including mismapping resulting from divergence, failure to account for gene duplication/extinction among species etc. On the other hand, if we attempt de novo assemblies, we vastly increase the computational complexity, presumably many genes won’t be assembled in many species, and thus won’t be analyzed, but we will probably get a more accurate quantification of those genes that are included.</p>

<p>Further complicating things, I’ve already tried mapping back to the reference genome with the Tuxedo pipeline, and mapping rates for species that diverge from the reference are stunningly low. Things get much better if I map to the transcriptome using a sensitive aligner like Stampy, but it seems we’ll lose some of the benefit of reference mapping if we have to use our established species-specific transcriptome.</p>

<p>I’ve been trying to review the literature, but there isn’t a whole lot on this issue. These these papers have been helpful in thinking about it though:</p>

<p>Vijay, Nagarjun, et al. <em>“Challenges and strategies in transcriptome assembly and differential gene expression quantification. A comprehensive in silico assessment of RNA‐seq experiments.”</em> Molecular ecology 22.3 (2013): 620-634.</p>

<p>Benjamin, Ashlee M., et al. <em>“Comparing reference-based RNA-Seq mapping methods for non-human primate data.”</em> BMC genomics 15.1 (2014): 570.</p>

<p>Hornett, Emily A., and Christopher W. Wheat. <em>“Quantitative RNA-Seq analysis in non-model species: assessing transcriptome assemblies as a scaffold and the utility of evolutionary divergent genomic reference species.”</em> BMC genomics 13.1 (2012): 361.</p>


</article>

<aside class="related">
  <h3>Related Posts</h3>
  <ul class="related-posts">
    
      <li>
        <h4>
          <a href="///2016/12/05/update/">
            Update.
            <small><time datetime="2016-12-05T00:00:00-05:00">05 Dec 2016</time></small>
          </a>
        </h4>
      </li>
    
      <li>
        <h4>
          <a href="///2016/03/05/codesnippets/">
            Testing code snippets.
            <small><time datetime="2016-03-05T00:00:00-05:00">05 Mar 2016</time></small>
          </a>
        </h4>
      </li>
    
      <li>
        <h4>
          <a href="///2016/03/02/newpaper/">
            New paper!
            <small><time datetime="2016-03-02T00:00:00-05:00">02 Mar 2016</time></small>
          </a>
        </h4>
      </li>
    
  </ul>
</aside>

		
<div id="disqus_thread"></div>
<script type="text/javascript">
    /* * * CONFIGURATION VARIABLES * * */
    var disqus_shortname = 'nreidgithubio';
    
    /* * * DON'T EDIT BELOW THIS LINE * * */
    (function() {
        var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
        dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript" rel="nofollow">comments powered by Disqus.</a></noscript>

      </main>



      <footer class="footer">
        <small>
          &copy; <time datetime="2017-10-26T12:57:39-04:00">2017</time>. All rights reserved.
        </small>
      </footer>
    </div>
  </body>
</html>
