<!DOCTYPE html>
<html lang="en">

  <head>
  <meta charset="UTF-8">
  <link href="http://gmpg.org/xfn/11" rel="profile">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Noah Reid &middot; 
    
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/poole.css">
  <link rel="stylesheet" href="/public/css/syntax.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/apple-touch-icon-precomposed.png">
  <link rel="shortcut icon" href="/public/favicon.ico">

  <!-- RSS -->
  <link rel="alternate" type="application/atom+xml" title="Noah Reid" href="/atom.xml">
</head>


  <body>

    <div class="container content">
      <header class="masthead">
        <h3 class="masthead-title">
          <a href="/" title="Home">Noah Reid</a>
          <small></small>

	  
	  	&nbsp;&nbsp;&nbsp;
	  	<small><a href="/about/">About</a></small>
	  
	  	&nbsp;&nbsp;&nbsp;
	  	<small><a href="/research/">Research</a></small>
	  
	  	&nbsp;&nbsp;&nbsp;
	  	<small><a href="/publications/">Publications</a></small>
	  
	  	&nbsp;&nbsp;&nbsp;
	  	<small><a href="/software/">Software</a></small>
	  
	  	&nbsp;&nbsp;&nbsp;
	  	<small><a href="/archive">Archive</a></small>
	  

        </h3>
      </header>

      <main>
        <div class="posts">
  
  <article class="post">
    <h2 class="post-title">
      <a href="/2016/03/05/codesnippets/">
        Testing code snippets.
      </a>
    </h2>

    <time datetime="2016-03-05T00:00:00-08:00" class="post-date">05 Mar 2016</time>

    <p>I’m just writing this to see how code snippets work using markdown and jekyll…</p>

<p>Here is a simple thing I often do to summarize distributions of pieces of information from the INFO field of a vcf file…</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">cl</span> <span class="o">&lt;-</span> <span class="s2">"tabix my.vcf.bgz Scaffold10 | grep -oP '(?&lt;=DP=)[^;]+'"</span>
<span class="n">dp</span> <span class="o">&lt;-</span> <span class="n">scan</span><span class="p">(</span><span class="n">pipe</span><span class="p">(</span><span class="n">cl</span><span class="p">))</span>
<span class="n">hist</span><span class="p">(</span><span class="n">dp</span><span class="p">,</span><span class="n">xlab</span><span class="o">=</span><span class="s2">"Depth"</span><span class="p">,</span><span class="n">breaks</span><span class="o">=</span><span class="m">100</span><span class="p">)</span></code></pre></figure>

<p>Here I’ve extracted the contents of the tag ‘DP’, which contains the depth of coverage of the variant.</p>

<p>In this case I’m feeding a line of code to the system, “cl”, with <code class="highlighter-rouge">pipe()</code>, and reading the results into R with <code class="highlighter-rouge">scan()</code>.</p>

<p>To break down “cl”: I use tabix to quickly access arbitrary chunks of VCF files, then pipe the output to grep. The flag <code class="highlighter-rouge">-o</code> extracts the regex match, and <code class="highlighter-rouge">-P</code> allows me to use what’s called a ‘positive lookahead’. That’s the section <code class="highlighter-rouge">(?&lt;=DP=)</code>. This requires the regex to match “DP=”, but doesn’t return it as part of the match. Then I match the characters in the field with <code class="highlighter-rouge">[^;]+</code>, which matches one or more of any character except “;”, which is the field delimiter.</p>

<p>Here’s the result:</p>

<p><img src="/assets/test.hist.png" alt="histogram" /></p>

  </article>
  
  <article class="post">
    <h2 class="post-title">
      <a href="/2016/03/02/newpaper/">
        New paper!
      </a>
    </h2>

    <time datetime="2016-03-02T00:00:00-08:00" class="post-date">02 Mar 2016</time>

    <p>I haven’t posted in a while, but the first of several papers from my postdoctoral work in the Whitehead lab has been <a href="http://bfg.oxfordjournals.org/content/early/2015/12/20/bfgp.elv060.abstract">published</a>!</p>

<p>Reid, Noah M., and Andrew Whitehead. <em>“Functional genomics to assess biological responses to marine pollution at physiological and evolutionary timescales: toward a vision of predictive ecotoxicology.”</em> <strong>Briefings in functional genomics</strong> (2015): elv060.</p>

  </article>
  
  <article class="post">
    <h2 class="post-title">
      <a href="/2015/03/03/rna-seq_issues/">
        Comparative RNA-seq
      </a>
    </h2>

    <time datetime="2015-03-03T00:00:00-08:00" class="post-date">03 Mar 2015</time>

    <p>I just dove into a really cool project looking at patterns of gene expression underlying osmotic adaptation in a phylogenetic comparative context. I think this work is really, really cool, but from the very start, we’re faced with problems that I’m not sure how to deal with.</p>

<p>The choice I’m currently mulling is whether we should try to assemble transcriptomes de novo for each species, or map them all back to a common reference. If we do reference mapping, things are computationally simpler and a larger gene set will ultimately be evaluated, but it seems that there must be some biases associated with the mapping, including mismapping resulting from divergence, failure to account for gene duplication/extinction among species etc. On the other hand, if we attempt de novo assemblies, we vastly increase the computational complexity, presumably many genes won’t be assembled in many species, and thus won’t be analyzed, but we will probably get a more accurate quantification of those genes that are included.</p>

<p>Further complicating things, I’ve already tried mapping back to the reference genome with the Tuxedo pipeline, and mapping rates for species that diverge from the reference are stunningly low. Things get much better if I map to the transcriptome using a sensitive aligner like Stampy, but it seems we’ll lose some of the benefit of reference mapping if we have to use our established species-specific transcriptome.</p>

<p>I’ve been trying to review the literature, but there isn’t a whole lot on this issue. These these papers have been helpful in thinking about it though:</p>

<p>Vijay, Nagarjun, et al. <em>“Challenges and strategies in transcriptome assembly and differential gene expression quantification. A comprehensive in silico assessment of RNA‐seq experiments.”</em> Molecular ecology 22.3 (2013): 620-634.</p>

<p>Benjamin, Ashlee M., et al. <em>“Comparing reference-based RNA-Seq mapping methods for non-human primate data.”</em> BMC genomics 15.1 (2014): 570.</p>

<p>Hornett, Emily A., and Christopher W. Wheat. <em>“Quantitative RNA-Seq analysis in non-model species: assessing transcriptome assemblies as a scaffold and the utility of evolutionary divergent genomic reference species.”</em> BMC genomics 13.1 (2012): 361.</p>


  </article>
  
</div>

<div class="pagination">
  
    <a class="pagination-item older" href="/page2">Older</a>
  
  
    <span class="pagination-item newer">Newer</span>
  
</div>

		
      </main>



      <footer class="footer">
        <small>
          &copy; <time datetime="2016-03-05T19:42:45-08:00">2016</time>. All rights reserved.
        </small>
      </footer>
    </div>
  </body>
</html>
